### DISCUSSION TITLE
A specific query disappears from the devtools, making it impossible to view, how do I bring it back?

### DISCUSSION THREAD

**Discussion by MaddoScientisto** [TanStack/query#5543] [Q&A] [answered]
I have a query that doesn't do a typical fetch request but instead it's a promise that sets an event listener, calls a function and waits for the event to be called to resolve the promise with the results, this is because the app is hosted on a winforms webview2 control and I'm interfacing with the interop.

This query for some reason is not showing up in the devtools anymore and it's been very frustrating because I need to debug it and I don't know what's making it disappear, I can see it briefly when the page first loads and then it's just gone, no other query does this.

[ Removed the code because it wasn't important to the question]

The query itself does work properly, it's just that it's gone from the devtools.
I checked the filters, cleared the cache multiple times, tried all the sorting options but that query still immediately deletes itself from the devtools, any idea why that would happen?

**Comment by TkDodo** [comment, marked as answer]
> any idea why that would happen?

do you have a stable QueryClient?

https://tkdodo.eu/blog/react-query-fa-qs#2-the-queryclient-is-not-stable

**Reply by MaddoScientisto** [reply]
Thank you, I'm going to check this out and report back

**Reply by MaddoScientisto** [reply]
That was exactly the reason! I Had an unstable queryclient, using an effect fixed it.
I'm going to cut the unnecessary code from the question, since I need to ask another one about a different problem but with almost the same code

**Reply by friedBrain123** [reply]
Your a legend thanks so much for this i was so confused move the query client in the App.tsx worked Straight away.

**Reply by KirschX** [reply]
I love you so much, really helped!

**Reply by 0mppula** [reply]
I love you so much thanks!!!

**Reply by atharva-bhange** [reply]
Thanks a lot this helped

**Reply by d12** [reply]
Thanks for this! @TkDodo Are there legitimate reasons to have multiple instances of the query client? I wonder if this should emit a warning in development to help others who run into this issue.

**Reply by TkDodo** [reply]
we have a lint rule for this:

https://tanstack.com/query/v5/docs/eslint/stable-query-client
